service/history/ndc/workflow_resetter.go

```diff
@@ -962,6 +962,9 @@ func reapplyEvents(
 				continue
 			}
 			attr := event.GetWorkflowExecutionTerminatedEventAttributes()
+			if attr.GetIdentity() == consts.IdentityHistoryService || attr.GetIdentity() == consts.IdentityResetter {
```

Comment:
actually shall we check the terminate reason? I think history service can terminate workflows for bunch of other reasons